cmd/prefetchfiles-nri-plugin/main.go

```diff
@@ -0,0 +1,272 @@
+			Destination: &args.PluginEvents,
+		},
+		&cli.StringFlag{
+			Name:        "server-path",
+			Value:       defaultServerPath,
+			Usage:       "the path of optimizer server binary",
+			Destination: &args.Config.ServerPath,
+		},
+		&cli.StringFlag{
+			Name:        "persist-dir",
+			Value:       defaultPersistDir,
+			Usage:       "the directory to persist accessed files list for container",
+			Destination: &args.Config.PersistDir,
+		},
+		&cli.IntFlag{
+			Name:        "timeout",
+			Value:       0,
+			Usage:       "the timeout to kill optimizer server, 0 to disable it",
+			Destination: &args.Config.Timeout,
+		},
+		&cli.BoolFlag{
+			Name:        "overwrite",
+			Usage:       "whether to overwrite the existed persistent files",
+			Destination: &args.Config.Overwrite,
+		},
+	}
+}
+
+func NewPluginFlags() *Flags {
+	var args PluginArgs
+	return &Flags{
+		Args: &args,
+		F:    buildFlags(&args),
+	}
+}
+
+type plugin struct {
+	stub stub.Stub
+	mask stub.EventMask
+}
+
+var (
+	cfg                  PluginConfig
+	log                  *logrus.Logger
+	logWriter            *syslog.Writer
+	globalFanotifyServer = make(map[string]*fanotify.Server)
+)
+
+func (p *plugin) Configure(config, runtime, version string) (stub.EventMask, error) {
+	log.Infof("got configuration data: %q from runtime %s %s", config, runtime, version)
+	if config == "" {
+		return p.mask, nil
+	}
+
+	tree, err := toml.Load(config)
+	if err != nil {
+		return 0, errors.Wrap(err, "parse TOML")
+	}
+	if err := tree.Unmarshal(&cfg); err != nil {
+		return 0, err
+	}
+
+	p.mask, err = api.ParseEventMask(cfg.Events...)
+	if err != nil {
+		return 0, errors.Wrap(err, "parse events in configuration")
+	}
+
+	log.Infof("configuration: %#v", cfg)
+
+	return p.mask, nil
+}
+
+func sendDataOverHTTP(data string, endpoint string) error {
+	url := "http://172.0.0.1:9110" + endpoint
+	req, err := http.NewRequest("POST",
+		url, bytes.NewBufferString(data))
+	if err != nil {
+		return err
+	}
+
+	conn, err := net.Dial("unix", "/run/containerd-nydus/system.sock")
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return conn, nil
+			},
+		},
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+func (p *plugin) RunPodSandbox(pod *api.PodSandbox) error {
+	name := pod.Name
+	parts := strings.Split(name, "-")
+	name = parts[0]
+	if pod.Annotations == nil {
+		log.Printf("error: Pod annotations is nil")
+		return nil
+	}
+	prefetchList, ok := pod.Annotations["prefetch_list"]
+	if !ok {
+		errMsg := "Pod.yaml annotations don't have prefetch list."
+		log.Printf("error: %s", errMsg)
+		return nil
+	}
+	msg := fmt.Sprintf("%s : %s", name, prefetchList)
+	err := sendDataOverHTTP(msg, "/api/v1/daemons/prefetch")
```

Comment:
It would be better to define this endpoint ("/api/v1/daemons/prefetch") as a constant.